src/windows/BarcodeScannerProxy.js

```diff
@@ -339,11 +350,31 @@ module.exports = {
             }
 
             focusControl.configure(focusConfig);
-            // Need to wrap this in setTimeout since continuous focus should start only after preview has started. See
-            // 'Remarks' at https://msdn.microsoft.com/en-us/library/windows/apps/windows.media.devices.focuscontrol.configure.aspx
-            return WinJS.Promise.timeout(200)
-            .then(function () {
-                return focusControl.focusAsync();
+
+            // Continuous focus should start only after preview has started. See 'Remarks' at 
+            // https://msdn.microsoft.com/en-us/library/windows/apps/windows.media.devices.focuscontrol.configure.aspx
+            return new WinJS.Promise(function (complete, error, progress) {
+                function onplaying() {
+                    capturePreview.removeEventListener("playing", onplaying, false);
+
+                    try {
```

Comment:
The semantics of `try/catch` block below is very similar to the code you have added to `focus` method. Could you factor this into separate method? Or reuse `focus` method here?